### ISSUE TITLE
Remove all annotations, visible and invisible

### ISSUE THREAD

**Issue by DavidTomic** [efremidze/Cluster#83]
### Issue Description
Hey,

Is there a way to remove all annotations.

Right now if I call `clusterManager.removeAll()` seems like only invisible annotations are removed.

In my app, I have searchBar which filters annotations, so I need them add/remove from map.

Thank you for your help.

### Environment

- **iOS Version**: [11.4.1]
- **Device(s)**: [iPhone 8]

EDIT:
Is it ok if I just add this line after calling `clusterManager.removeAll()`:
`mapView.removeAnnotations(mapView.annotations)`

**Comment by svitzebec** [comment]
I can confirm this, saw some similar behaviour in my case.

When updating the map I did something like this:
```
clusterManager.removeAll()
clusterManager.add(newAnnotations)
clusterManager.reload(mapView: mapView)
```
This led to some buggy behaviour, where too many annotations were shown on the map view.

Then I tried the solution @DavidTomic suggested (adding `mapView.removeAnnotations(mapView.annotations)` after the `removeAll` call) and everything started working as it should.

So the updated code in my case looks like:
```
clusterManager.removeAll()
mapView.removeAnnotations(mapView.annotations)
clusterManager.add(newAnnotations)
clusterManager.reload(mapView: mapView)
```

My guess would be that there is some issue in the `clusteredAnnotations(zoomScale: Double, visibleMapRect: MKMapRect, operation: Operation? = nil)` function, specifically when obtaining the annotations to remove. But I did not manage to figure out exactly where the problem is.

**Comment by efremidze** [comment]
Thanks, I'll take a look asap.

**Comment by sebleclerc** [comment]
I do have the same problem. I have something similar to @svitzebec and using suggested answer from @DavidTomic, it's OK right now!

P.S. awesome library!

**Comment by efremidze** [comment]
I'll pushed a fix in [2.2.8](https://github.com/efremidze/Cluster/releases/tag/2.2.8). Let me know if you find any issues.

**Comment by svitzebec** [comment]
@efremidze I just updated to 2.2.8 and tested the fix in my use case. The issue seems to be fixed.
Great job and thanks!

**Comment by edwardmp** [comment]
I'm not sure if this is related to this issue, but I just tried this framework and the clustering behavior is really odd. Some clusters keep appearing even when zooming in displaying a count higher than the total number of pins.

**Comment by efremidze** [comment]
@edwardmp I'll take a look

**Comment by efremidze** [comment]
@edwardmp That may be related to the `cellSize`. I'm currently adding more options to adjust `cellSize`.

**Comment by efremidze** [comment]
Released [2.3.0](https://github.com/efremidze/Cluster/releases/tag/2.3.0). Let me know if you still see the bug.
